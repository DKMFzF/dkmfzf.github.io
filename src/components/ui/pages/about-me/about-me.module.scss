@use '~styles/variables' as *;
@use '~styles/mixins' as *;

// control-section-var
$img-side-size: 280px;
$img-border-radius: 50%;
$padding-card-block: 40px;
$border-block: 1px solid $bg-primary-color;
$card-min-w: 360px;
$font-size-sub-title: 13px;
$font-size-title-page: 150px;
$font-weight-title: 900;
$premary-card-color: #cec2e6;

// grid-settings
$grid-gap: 20px;
$grid-rows: auto auto auto;
$grid-columns: repeat(4, 1fr);
$title-block-row: 1;
$title-block-col: 1 / span 4;
$stack-block-row: 2 / span 2;
$stack-block-col: 1;
$timeline-block-row: 2 / span 2;
$timeline-block-col: 2;
$title-block-row: 1;
$title-block-col: 1 / span 4;

.aboutMe {
	// global-setting-card

	&__content {
		display: grid;
		grid-template-rows: $grid-rows;
		grid-template-columns: $grid-columns;
		gap: $grid-gap;
		block-size: 100%;
	}

	&__cards {
		min-width: $card-min-w;
	}

	// title-card

	&__titleContainer {
		grid-row: $title-block-row;
		grid-column: $title-block-col;
	}

	&__title {
		font-weight: $font-weight-title;
		font-size: $font-size-title-page;
		font-family: $font-title;
		text-transform: uppercase;
	}

	// stack-card

	&__stack {
		position: relative;
		grid-row: 2 / span 2;
		grid-column: 1;
		align-content: normal;
		color: white;
		background-color: $bg-primary-color;
	}

	// timeline-card

	&__timeline {
		grid-row: $timeline-block-row;
		grid-column: $timeline-block-col;
		background-color: $premary-card-color;
		border: 1px solid $premary-card-color;
	}

	// other-card

	&__info3 {
		grid-row: 2;
		grid-column: 3 / span 2;
	}

	&__info4 {
		grid-row: 3;
		grid-column: 3 / span 2;
	}

	&__imageContainer {
		display: flex;
		grid-row: 1 / span 3;
		grid-column: 4;
		justify-content: center;
		align-items: center;
	}

	&__description {
		margin-bottom: 0;
		font-size: $font-size-min;
		line-height: $section-description-line-height;
		opacity: $opacity;
	}

	&__image {
		width: $img-side-size;
		height: $img-side-size;
		object-fit: cover;
		border-radius: $img-border-radius;
	}
}
