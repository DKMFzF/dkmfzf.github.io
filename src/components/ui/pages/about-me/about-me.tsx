import { FC } from 'react';
import myImage from '@images/my-photo.jpg';

import styles from './about-me.module.scss';
import { CommonPage } from '@pages';
import { AboutMeArticle, ExperienceCard, KnowledgeSidebar } from '@components';

export const AboutMePageUI: FC = () => (
	<CommonPage pageStyles={styles['about-me']}>
		<div className={styles['about-me__content']}>
			<AboutMeArticle
				className={styles['about-me__title-container']}
				borderFlag
				borderRadiusFlag
			>
				<h1 className={styles['about-me__title']}>About Me</h1>
			</AboutMeArticle>

			<AboutMeArticle
				subtitle='my tech skills'
				className={`${styles['about-me__stack']} ${styles['about-me__cards']}`}
				borderFlag
				borderRadiusFlag
				flexFlag
			>
				<KnowledgeSidebar />
			</AboutMeArticle>

			<AboutMeArticle
				subtitle='my timeline'
				className={`${styles['about-me__cards']} ${styles['about-me__timeline']}`}
				borderFlag
				borderPrimaryColorFlag
				borderRadiusFlag
				flexFlag
			>
				<ExperienceCard />
			</AboutMeArticle>

			<AboutMeArticle
				className={styles['about-me__image-dkmfzf']}
				paddingFlag
			>
				<div className={styles['about-me__image-wrapper']}>
					<img
						src={myImage}
						alt='human user'
						className={styles['about-me__image']}
					/>
				</div>
			</AboutMeArticle>

			<AboutMeArticle
				className={styles['about-me__discriptrion-dkmfzf']}
				borderFlag
				borderRadiusFlag
			>
				<p className={styles['about-me__description']}>
					I am in love with complex algorithmic programming tasks and
					Hackathons!
				</p>
			</AboutMeArticle>
		</div>
	</CommonPage>
);
