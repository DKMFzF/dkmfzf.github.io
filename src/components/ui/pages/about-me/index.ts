export { AboutMePageUI } from './about-me';
