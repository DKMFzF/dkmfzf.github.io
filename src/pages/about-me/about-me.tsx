import { AboutMePageUI } from '@ui/index';

export const AboutMePage = () => <AboutMePageUI />;
