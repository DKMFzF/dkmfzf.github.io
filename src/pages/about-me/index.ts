export { AboutMePage } from './about-me';
