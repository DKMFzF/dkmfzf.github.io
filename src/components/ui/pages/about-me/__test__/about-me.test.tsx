import { AboutMePageUI } from '../about-me';
import { HelmetProvider } from 'react-helmet-async';
import { testRenderComponent } from '@utils-testing';

jest.mock(
	'../../../../../../public/kirill-doroshev-resume.pdf',
	() => 'test-resume.pdf'
);

describe('[AboutMePageUI]', () => {
	it('Renders about-me-page correctly and matches snapshot', () =>
		testRenderComponent(
			<HelmetProvider>
				<AboutMePageUI />
			</HelmetProvider>
		));
});
