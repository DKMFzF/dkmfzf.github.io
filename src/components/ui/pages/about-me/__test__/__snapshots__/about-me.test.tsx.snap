// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AboutMePageUI] Renders about-me-page correctly and matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section
      class="main__section-bg about-me"
    >
      <div
        class="about-me__content"
      >
        <article
          class="about-me__block about-me__title-container 
		
		about-me__padding-default
		about-me__border-radius-default
		about-me__border-flag
		"
        >
          <h1
            class="about-me__title"
          >
            About Me
          </h1>
        </article>
        <article
          class="about-me__block about-me__stack about-me__cards 
		
		about-me__padding-default
		about-me__border-radius-default
		about-me__border-flag
		about-me__flex-flag"
        >
          <header
            class="about-me__sub-title"
          >
            <h3>
              my tech skills
            </h3>
          </header>
          <div
            class="knowledge-sidebar"
          >
            <div
              class="knowledge-sidebar__body-slider"
            >
              <button
                aria-label="Previous card"
                class="knowledge-sidebar__button"
                data-cy="card-btn-left"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 18L9 12L15 6"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
              <div
                aria-live="polite"
                class="knowledge-sidebar__content"
              >
                <div
                  class="knowledge-sidebar__card fade-in"
                >
                  <span
                    class="knowledge-sidebar__skill"
                    data-cy="card-title"
                  >
                    React
                  </span>
                </div>
              </div>
              <button
                aria-label="Next card"
                class="knowledge-sidebar__button"
                data-cy="card-btn-right"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 18L15 12L9 6"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
            <footer
              aria-live="polite"
              class="knowledge-sidebar__stack-discription"
            >
              <span
                class="fade-in"
              >
                Experience in SPA development. I work confidently with hooks, components, and routing.
              </span>
              <div
                class="knowledge-sidebar__stack-arrow"
              >
                &gt;
              </div>
            </footer>
          </div>
        </article>
        <article
          class="about-me__block about-me__cards about-me__timeline 
		about-me__border-primaty
		about-me__padding-default
		about-me__border-radius-default
		about-me__border-flag
		about-me__flex-flag"
        >
          <header
            class="about-me__sub-title"
          >
            <h3>
              my timeline
            </h3>
          </header>
          <div
            class="experience-card"
          >
            <div
              class="experience-card__experience-container"
            >
              <span
                class="experience-card__number"
              >
                2
              </span>
              <span
                class="experience-card__years-label"
              >
                years
              </span>
            </div>
            <div
              class="experience-card__container-footer"
            >
              <p
                class="experience-card__description"
              >
                I am immersed in the IT community, participate in hackathons, take courses and study at the SFU Institute
              </p>
              <div
                class="experience-card__timeline-arrow"
              >
                &gt;
              </div>
            </div>
          </div>
        </article>
        <article
          class="about-me__block about-me__image-dkmfzf 
		
		
		
		
		"
        >
          <div
            class="about-me__image-wrapper"
          >
            <img
              alt="human user"
              class="about-me__image"
              src=""
            />
          </div>
        </article>
        <article
          class="about-me__block about-me__discriptrion-dkmfzf 
		
		about-me__padding-default
		about-me__border-radius-default
		about-me__border-flag
		"
        >
          <p
            class="about-me__description"
          >
            I am in love with complex algorithmic programming tasks and Hackathons!
          </p>
        </article>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
